sql/src/main/java/io/crate/expression/scalar/string/StringPaddingFunction.java

```diff
@@ -0,0 +1,136 @@
+import io.crate.expression.scalar.ThreeParametersFunction;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+
+import java.util.List;
+import java.util.Locale;
+
+public class StringPaddingFunction extends Scalar<String, Object> {
+
+    public static final String LNAME = "lpad";
+    public static final String RNAME = "rpad";
+    public static final String DEFAULT_FILL = " ";
+    public static final int LEN_LIMIT = 1000;
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(LNAME, new Resolver(LNAME, StringPaddingFunction::lpad));
+        module.register(RNAME, new Resolver(RNAME, StringPaddingFunction::rpad));
+    }
+
+    private final FunctionInfo info;
+    private final ThreeParametersFunction<char[], Integer, char[], String> func;
+
+    private StringPaddingFunction(FunctionInfo info, ThreeParametersFunction<char[], Integer, char[], String> func) {
+        this.info = info;
+        this.func = func;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public String evaluate(TransactionContext txnCtx, Input[] args) {
+        assert args.length == 2 || args.length == 3 : String.format(
+            Locale.ENGLISH,
+            "number of arguments must be 2 (optionally 3), got %d instead",
+            args.length);
+
+        String str = (String) args[0].value();
+        Number len = (Number) args[1].value();
+        String fill = args.length == 3 ? (String) args[2].value() : DEFAULT_FILL;
+        if (str == null || len == null || fill == null) {
+            return null;
+        }
+        int lenValue = len.intValue();
+        if (lenValue > LEN_LIMIT) {
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                                                             "len argument exceeds predefined limit of %d",
+                                                             LEN_LIMIT));
+        }
+
+        if (lenValue <= 0 || (str.isEmpty() && fill.isEmpty())) {
+            return "";
+        } else if (str.length() >= lenValue) {
+            return str.substring(0, lenValue);
+        } else if (fill.isEmpty()) {
+            return str;
+        }
+        return func.apply(str.toCharArray(), lenValue, fill.toCharArray());
+    }
+
+    private static String lpad(char[] srcChars, int len, char[] fillChars) {
+        char[] buffer = new char[len];
+        int padLen = len - srcChars.length;
+        System.arraycopy(srcChars, 0, buffer, padLen, srcChars.length);
+        for (int i = 0; i < padLen; i++) {
```

Comment:
srcChars.length is already written, so we could probably start from there instead of from 0?

Reply:
In there I copy str to the buffer, and then I take as many chars from fill as needed to fill up the buffer cycling through the fill chars. So at first I populate the right hand side of the buffer and then I fill the left, so I have to start at 0.

Reply:
Oh right, somehow I mixed up lpad with rpad.